openedx/features/course_experience/tests/views/test_course_home.py

```diff
@@ -340,54 +346,86 @@ def test_course_does_not_expire_for_different_roles(self):
-            user = UserFactory.create(password=self.TEST_PASSWORD)
-            enrollment = CourseEnrollment.enroll(user, course.id, mode=CourseMode.AUDIT)
-            course_team_member['course_role'](course.id).add_users(user)
-            ScheduleFactory(start=THREE_YEARS_AGO, enrollment=enrollment)
-            users.append({
-                'user': user,
-                'description': course_team_member['description'],
-            })
-
-        # ensure that all users who should have access indefinitely do
-        for user in users:
-            self.client.login(username=user['user'].username, password=self.TEST_PASSWORD)
-            response = self.client.get(url)
-            self.assertEqual(
-                response.status_code,
-                200,
-                "Should not expire access for user [{}]".format(user['description'])
-            )
+        user = UserFactory.create(password=self.TEST_PASSWORD)
+        ScheduleFactory(
+            start=THREE_YEARS_AGO,
+            enrollment__mode=CourseMode.VERIFIED,
+            enrollment__course_id=course.id,
+            enrollment__user=user
+        )
+
+        # ensure that the user who has indefinite access
+        self.client.login(username=user.username, password=self.TEST_PASSWORD)
+        response = self.client.get(url)
+        self.assertEqual(
+            response.status_code,
+            200,
+            "Should not expire access for user",
+        )
+
+    @mock.patch.dict(settings.FEATURES, {'DISABLE_START_DATES': False})
+    @ddt.data(
+        InstructorFactory,
+        StaffFactory,
+        BetaTesterFactory,
+        OrgStaffFactory,
+        OrgInstructorFactory,
+    )
+    def test_course_does_not_expire_for_course_roles(self, role_factory):
+        """
+        There are a number of different roles/users that should not lose access after the expiration date.
+        Ensure that users who should not lose access get a 200 (ok) response
+        when attempting to visit the course after their would be expiration date.
+        """
+        course = CourseFactory.create(start=THREE_YEARS_AGO)
+        url = course_home_url(course)
+
+        user = role_factory.create(password=self.TEST_PASSWORD, course_key=course.id)
+        ScheduleFactory(
+            start=THREE_YEARS_AGO,
+            enrollment__mode=CourseMode.AUDIT,
+            enrollment__course_id=course.id,
+            enrollment__user=user
+        )
+
+        # ensure that the user who has indefinite access
+        self.client.login(username=user.username, password=self.TEST_PASSWORD)
+        response = self.client.get(url)
+        self.assertEqual(
+            response.status_code,
+            200,
+            "Should not expire access for user",
+        )
+
+    @mock.patch.dict(settings.FEATURES, {'DISABLE_START_DATES': False})
+    @ddt.data(
+        GlobalStaffFactory,
+    )
+    def test_course_does_not_expire_for_global_users(self, role_factory):
```

Comment:
can test_course_does_not_expire_for_global_users be combined with test_course_does_not_expire_for_course_roles?

Reply:
nah that's okay, I couldn't see what the difference was between the methods.  If you want to do that you can, but there is no need to over engineer it.  Either way is fine with me!